.github/actions/setup-testground/action.yml

```diff
@@ -19,8 +19,13 @@ runs:
       if: ${{ inputs.testground_ref == 'edge' }}
       shell: bash
       run: |
-        /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/testground/testground/master/install.sh)"
-
+        for i in 1 2 3; do
+          echo "=== Attempt $i ==="
+          /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/testground/testground/master/install.sh)" && \
```

Comment:
Now that there are retries we could download the shell script ahead of the loop.